src/libraries/System.Security.Cryptography.X509Certificates/tests/RevocationTests/DynamicRevocationTests.cs

```diff
@@ -990,6 +990,95 @@ public static void TestRevocationWithNoNextUpdate_Revoked()
                 });
         }
 
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.OSX)] //macOS does not support offline chain building.
+        public static void TestRevocation_Offline_NotRevoked()
+        {
+            SimpleTest(
+                PkiOptions.CrlEverywhere,
+                (root, intermediate, endEntity, onlineHolder) =>
+                {
+                    using ChainHolder offlineHolder = new ChainHolder();
+                    X509Chain offlineChain = offlineHolder.Chain;
+                    X509Chain onlineChain = onlineHolder.Chain;
+                    CopyChainPolicy(from: onlineChain, to: offlineChain);
+                    offlineChain.ChainPolicy.RevocationMode = X509RevocationMode.Offline;
+
+                    SimpleRevocationBody(
+                        onlineHolder,
+                        endEntity,
+                        rootRevoked: false,
+                        issrRevoked: false,
+                        leafRevoked: false);
+
+                    SimpleRevocationBody(
```

Comment:
Ideally we would shut down the responder between these two calls.

The easiest way is to probably do something like

* add

```C#
private delegate void RunSimpleTestWithResponder(
            RevocationResponder responder,
            CertificateAuthority root,
            CertificateAuthority intermediate,
            X509Certificate2 endEntity,
            ChainHolder chainHolder);
```

Then make there be three overloads of SimpleTest:

```C#
        private static void SimpleTest(
            PkiOptions pkiOptions,
            RunSimpleTest callback,
            [CallerMemberName] string callerName = null,
            bool pkiOptionsInTestName = true) =>
                SimpleTest(pkiOptions, callback, null, callerName, pkiOptionsInTestName);

        private static void SimpleTest(
            PkiOptions pkiOptions,
            RunSimpleTestWithResponder callback,
            [CallerMemberName] string callerName = null,
            bool pkiOptionsInTestName = true) =>
                SimpleTest(pkiOptions, null, callback, callerName, pkiOptionsInTestName);

        private static void SimpleTest(
            PkiOptions pkiOptions,
            RunSimpleTest? callback,
            RunSimpleTestWithResponder? callbackWithResponder,
            string callerName,
            bool pkiOptionsInTestName)
        {
            Assert.True((callback == null) != (callbackWithResponder == null), "Caller provided exactly one callback option");

            // existing body, adjusted to use whichever callback it got.
            ...
```

Reply:
Editing every existing test to be able to adjust the delegate also works, of course :smile:.